### ISSUE TITLE
Allow specifying local cabal options

### ISSUE THREAD

**Issue by RyanGlScott** [haskell-CI/haskell-ci#129]
Just as there is currently a `--local-ghc-options` flag for adding `ghc-options` to the autogenerated `cabal.project` file, it would be convenient to have something similar for `flags`. I currently have to add this by hand to `free`'s `.travis.yml`, for instance:

https://github.com/ekmett/free/blob/26b5a9b947969ff85e419443ae11e26319693a00/.travis.yml#L106-L108

**Comment by RyanGlScott** [comment]
Or perhaps we need a more general mechanism for adding stanzas to a `package` field in the auto-generated `cabal.project` file. (For instance, I just found myself needing to generate `library-coverage: true`.)

**Comment by RyanGlScott** [comment]
I think what might be nicer is to allow users to write stub files (lousy name proposal: `cabal.project.stub`) with contents like:

```
package my-package
  library-coverage: true
```

And have `haskell-ci` append that to the end of its auto-generated `cabal.project` files. This way, we wouldn't need an abundance of flags to support every possible `Cabal` flag.

**Comment by RyanGlScott** [comment]
https://github.com/haskell-CI/haskell-ci/pull/157 added constraint sets via `.haskell-ci` files. However, it's unclear to me what the relationship between that and this idea is (for instance, I'm not sure if I can use constraint sets to achieve what I want here).

**Comment by phadej** [comment]
This will be resolved once next `Cabal` version is released. Hopefully soon.

**Comment by RyanGlScott** [comment]
This was addressed in #129 with the addition of `raw-project` stanzas, which let you do things like this:

```
raw-project
  package my-package
    library-coverage: true
```
